lib/nebulex/caching/decorators.ex

```diff
@@ -257,11 +257,10 @@ defmodule Nebulex.Caching.Decorators do
       else
         value = unquote(block)
 
-        if apply(match, [value]) do
-          cache.set(key, value, opts)
-        else
-          value
+        cond do
```

Comment:
Let's keep the previous `if` approach here since `credo` complains about it.

```elixir
if apply(match, [value]) do
  cache.set(key, value, opts)
else
  value
end
```

```
┃ [F] → Cond statements should contain at least two conditions besides `true`, 
┃       consider using `if` instead.
┃       lib/nebulex/caching/decorators.ex:289:7 #(Nebulex.Caching.Decorators.action_logic)
┃ [F] → Cond statements should contain at least two conditions besides `true`, 
┃       consider using `if` instead.
┃       lib/nebulex/caching/decorators.ex:260:9 #(Nebulex.Caching.Decorators.action_logic)
```

Reply:
The problem with it @cabol, is that the fix is exactly for that. I'll test this workaround:

```elixir
should_update = apply(match, [value])
cond do
  should_update == true -> cache.set(key, value, opts)
  should_update == false -> value
end
```

Reply:
Got it! I tried this and did work:

```
cond do
  apply(match, [value]) -> cache.set(key, value, opts)
  true -> value
end
```

Reply:
That's odd, my credo complained. Even my credo layer on emacs did also. =/

![Screenshot from 2020-06-11 12-31-14](https://user-images.githubusercontent.com/8040844/84406169-8a071c80-abdf-11ea-9ffb-c3aff288a6ed.png)

I'm trying something clever with pattern matching here. I think its going to be better

Reply:
Okay, all good! The pattern match was clever enough to not break Dialyzer and Credo. And is also a cleaner code. I'll submit now.

I also did the same fix to that I did on `key` on `match`, since there might be a corner case that has the same behavior that I was trying to avoid on using `Kernel.||/2`. 

cc @cabol

Reply:
> Got it! I tried this and did work:
> 
> ```elixir
> cond do
>   apply(match, [value]) ->
>     cache.set(key, value, opts)
> 
>   true ->
>     value
> end
> ```
> 
> And credo didn't complain either

Okay, I'll stop fighting metaprogramming since apparently I know less than I expected. Tested with `with` statement and I works. The IR code is: 

```erlang
          case erlang:apply(__@5, [__@11]) of
            true -> __@14 = __@1:set(__@2, __@11, __@4), __@14;
            __@12 ->
                case __@12 of
                  false -> __@11;
                  __@13 -> erlang:error({with_clause, __@13})
                end
          end;
```

It works because it dependes on exec time to figure stuff out! :D

cc @cabol

Reply:
> Got it! I tried this and did work:
> 
> ```elixir
> cond do
>   apply(match, [value]) ->
>     cache.set(key, value, opts)
> 
>   true ->
>     value
> end
> ```
> 
> And credo didn't complain either

Okay, I'll stop fighting metaprogramming since apparently I know less than I expected. Tested with `with` statement and I works. The IR code is: 

```erlang
          case erlang:apply(__@5, [__@11]) of
            true -> __@14 = __@1:set(__@2, __@11, __@4), __@14;
            __@12 ->
                case __@12 of
                  false -> __@11;
                  __@13 -> erlang:error({with_clause, __@13})
                end
          end;
```

It works because it dependes on exec time to figure stuff out! :D

cc @cabol